File: server/src/models/Event.php
<?php
/**
 * Created by PhpStorm.
 * User: Jacek
 * Date: 17.11.2017
 * Time: 12:03
 */

namespace src\models;


class Event implements \JsonSerializable
{
    private $competitionId;
    private $competitionName;
    private $competitionDescription;
    private $competitionDate;

    private $mainReferee;
    private $rater;
    private $typeOfCompetition;

    private $price;
    private $thumbnailUrl;

    /**
     * Event constructor.
     */
    public function __construct()
    {
    }

    /**
     * @return mixed
     */
    public function getCompetitionId()
    {
        return $this->competitionId;
    }

    /**
     * @param mixed $competitionId
     */
    public function setCompetitionId($competitionId)
    {
        $this->competitionId = $competitionId;
    }


    /**
     * @return mixed
     */
    public function getCompetitionName()
    {
        return $this->competitionName;
    }

    /**
     * @param mixed $competitionName
     */
    public function setCompetitionName($competitionName)
    {
        $this->competitionName = $competitionName;
    }

    /**
     * @return mixed
     */
    public function getCompetitionDescription()
    {
        return $this->competitionDescription;
    }

    /**
     * @param mixed $competitionDescription
     */
    public function setCompetitionDescription($competitionDescription)
    {
        $this->competitionDescription = $competitionDescription;
    }

    /**
     * @return mixed
     */
    public function getCompetitionDate()
    {
        return $this->competitionDate;
    }

    /**
     * @param mixed $competitionDate
     */
    public function setCompetitionDate($competitionDate)
    {
        $this->competitionDate = $competitionDate;
    }

    /**
     * @return mixed
     */
    public function getPrice()
    {
        return $this->price;
    }

    /**
     * @param mixed $price
     */
    public function setPrice($price)
    {
        $this->price = $price;
    }

    /**
     * @return mixed
     */
    public function getThumbnailUrl()
    {
        return $this->thumbnailUrl;
    }

    /**
     * @param mixed $thumbnailUrl
     */
    public function setThumbnailUrl($thumbnailUrl)
    {
        $this->thumbnailUrl = $thumbnailUrl;
    }


    /**
     * @return MainReferee
     */
    public function getMainReferee()
    {
        return $this->mainReferee;
    }

    /**
     * @param MainReferee $mainReferee
     */
    public function setMainReferee(MainReferee $mainReferee)
    {
        $this->mainReferee = $mainReferee;
    }

    /**
     * @return Rater
     */
    public function getRater()
    {
        return $this->rater;
    }

    /**
     * @param Rater $rater
     */
    public function setRater(Rater $rater)
    {
        $this->rater = $rater;
    }

    /**
     * @return TypeOfCompetition
     */
    public function getTypeOfCompetition()
    {
        return $this->typeOfCompetition;
    }

    /**
     * @param TypeOfCompetition $typeOfCompetition
     */
    public function setTypeOfCompetition(TypeOfCompetition $typeOfCompetition)
    {
        $this->typeOfCompetition = $typeOfCompetition;
    }

    public function jsonSerialize() {
        return [
            'idCompetition' => $this->competitionId,
            'competitionName' => $this->competitionName,
            'competitionDescription' => $this->competitionDescription,
            'competitionDate' => $this->competitionDate,
            'thumbnailUrl' => $this->thumbnailUrl,
            'price' => $this->price,

            'mainReferee' => $this->mainReferee,
            'rater' => $this->rater,
            'typeOfCompetition' => $this->typeOfCompetition
        ];
    }


}

File: client/android/app/src/main/java/com/politechnika/shootingrange/models/Event.java
package com.politechnika.shootingrange.models;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;
import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;

/**
 * Created by Jacek on 16.11.2017.
 */

public class Event implements Serializable {
    public static final String TAG_ID_COMPETITION = "idCompetition";
    public static final String TAG_CLICKED_EVENT = "clickedEvent";
    public static final String TAG_THUMBNAIL_URL = "thumbnailUrl";
    public static final String TAG_EVENT_NAME = "competitionName";
    public static final String TAG_EVENT_DESCRIPTION = "competitionDescription";
    public static final String TAG_EVENT_DATE = "competitionDate";
    public static final String TAG_PRICE = "price";
    public static final String TAG_NUMBER_OF_COMPETITORS = "numberOfCompetitors";

    @Override
    public String toString() {
        return String.format("Event#idCompetition: %s, competitionName: %s, competitionDescription: %s, competitionDate: %s, price: %s, numberOfCompetitors: %s, thumbnailUrl: %s\n" +
                "mainReferee: %s\n" +
                "rater: %s\n" +
                "typeOfCompetition: %s\n", idCompetition, competitionName, competitionDescription, competitionDate, price, numberOfCompetitors, thumbnailUrl, mainReferee, rater, typeOfCompetition);
    }
    @SerializedName(TAG_ID_COMPETITION)
    private String idCompetition;
    @SerializedName(TAG_EVENT_NAME)
    private String competitionName;
    @SerializedName(TAG_EVENT_DESCRIPTION)
    private String competitionDescription;
    @SerializedName(TAG_EVENT_DATE)
    private String competitionDate;

    @SerializedName("mainReferee")
    private MainReferee mainReferee;
    @SerializedName("rater")
    private Rater rater;
    @SerializedName("typeOfCompetition")
    private TypeOfCompetition typeOfCompetition;

    @SerializedName("price")
    private double price;
    @SerializedName("numberOfCompetitors")
    private int numberOfCompetitors;
    @SerializedName("thumbnailUrl")
    private String thumbnailUrl;

    // determines whether event is available or unavailable for user
    private boolean isAvailable;

    public Event() {

    }

    public String getCompetitionName() {
        return competitionName;
    }

    public void setCompetitionName(String competitionName) {
        this.competitionName = competitionName;
    }

    public String getCompetitionDescription() {
        return competitionDescription;
    }

    public void setCompetitionDescription(String competitionDescription) {
        this.competitionDescription = competitionDescription;
    }

    public String getCompetitionDate() {
        return competitionDate;
    }

    public void setCompetitionDate(String competitionDate) {
        this.competitionDate = competitionDate;
    }

    public double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    public int getNumberOfCompetitors() {
        return numberOfCompetitors;
    }

    public void setNumberOfCompetitors(int numberOfCompetitors) {
        this.numberOfCompetitors = numberOfCompetitors;
    }

    public String getIdCompetition() {
        return idCompetition;
    }

    public void setIdCompetition(String idCompetition) {
        this.idCompetition = idCompetition;
    }

    public MainReferee getMainReferee() {
        return mainReferee;
    }

    public void setMainReferee(MainReferee mainReferee) {
        this.mainReferee = mainReferee;
    }

    public Rater getRater() {
        return rater;
    }

    public void setRater(Rater rater) {
        this.rater = rater;
    }

    public TypeOfCompetition getTypeOfCompetition() {
        return typeOfCompetition;
    }

    public void setTypeOfCompetition(TypeOfCompetition typeOfCompetition) {
        this.typeOfCompetition = typeOfCompetition;
    }

    public String getThumbnailUrl() {
        return thumbnailUrl;
    }

    public void setThumbnailUrl(String thumbnailUrl) {
        this.thumbnailUrl = thumbnailUrl;
    }


    public boolean isAvailable() {
        return isAvailable;
    }

    public void setAvailable(boolean available) {
        isAvailable = available;
    }


    public boolean isOutdated() {
        Date competitionParsedDate = new Date();
        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
        try {
            competitionParsedDate = sdf.parse(competitionDate);
        } catch (ParseException e) {
            e.printStackTrace();
        }

        if (new Date().before(competitionParsedDate)){
            return false;
        }
        else {
            return true;
        }
    }

}
